server/src/main/java/com/microsoft/java/bs/core/internal/utils/JavaUtils.java

```diff
@@ -0,0 +1,79 @@
+package com.microsoft.java.bs.core.internal.utils;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.lang.Runtime.Version;
+
+/**
+ * Utility class for Java.
+ */
+public class JavaUtils {
+
+  public JavaUtils() {
+  }
+
+  /**
+   * Reads the Java Version from the given JDK file.
+   *
+   * @param jdkFile JDK file to read the java version from.
+   * @return Java Version of the given JDK or null upon failure.
+   */
+  public static String getJavaVersionFromFile(File jdkFile) throws IOException {
+
+    ProcessBuilder processBuilder =
+        new ProcessBuilder(jdkFile.getAbsolutePath() + "/bin/java", "-version");
```

Comment:
what will happen if the executable file does not exist?

Reply:
That should ideally throw an `IOException` when we try to start the process using `processBuilder.start()`. If the exception occurs it will be up to the class calling this utility function on how to handle it.

I think we can have more unit tests for this method, right we are only simulating the error in `testGetJavaVersionFromFileForException`.

Reply:
Ok, let's add one for file not exist case

Reply:
Added in commit [1e1ad04](https://github.com/microsoft/build-server-for-gradle/pull/165/commits/1e1ad040873483729109c47bc7f489b24f0afa67)